driver-sync/src/test/functional/com/mongodb/client/RetryableWritesProseTest.java

```diff
@@ -179,6 +188,74 @@ public static <R> void poolClearedExceptionMustBeRetryable(
+        assumeTrue(serverVersionAtLeast(6, 0) && isDiscoverableReplicaSet());
+        ServerAddress primaryServerAddress = Fixture.getPrimary();
+        CompletableFuture<FailPoint> futureFailPointFromListener = new CompletableFuture<>();
+        CommandListener commandListener = new CommandListener() {
+            private final AtomicBoolean configureFailPoint = new AtomicBoolean(true);
+
+            @Override
+            public void commandSucceeded(final CommandSucceededEvent event) {
+                if (event.getCommandName().equals("insert") && configureFailPoint.compareAndSet(true, false)) {
+                    Assertions.assertTrue(futureFailPointFromListener.complete(FailPoint.enable(
+                            new BsonDocument()
+                                    .append("configureFailPoint", new BsonString("failCommand"))
+                                    .append("mode", new BsonDocument()
+                                            .append("times", new BsonInt32(1)))
+                                    .append("data", new BsonDocument()
+                                            .append("failCommands", new BsonArray(singletonList(new BsonString("insert"))))
+                                            .append("errorCode", new BsonInt32(10107))
+                                            .append("errorLabels", new BsonArray(Stream.of("RetryableWriteError", "NoWritesPerformed")
+                                                    .map(BsonString::new).collect(Collectors.toList())))),
+                            primaryServerAddress
+                    )));
+                }
+            }
+        };
+        BsonDocument failPointDocument = new BsonDocument()
+                .append("configureFailPoint", new BsonString("failCommand"))
+                .append("mode", new BsonDocument()
+                        .append("times", new BsonInt32(1)))
+                .append("data", new BsonDocument()
+                        .append("writeConcernError", new BsonDocument()
+                                .append("code", new BsonInt32(91))
+                                .append("errorLabels", new BsonArray(Stream.of("RetryableWriteError")
+                                        .map(BsonString::new).collect(Collectors.toList()))))
+                        .append("failCommands", new BsonArray(singletonList(new BsonString("insert")))));
+        try (MongoClient client = clientCreator.apply(getMongoClientSettingsBuilder()
+                .retryWrites(true)
+                .addCommandListener(commandListener)
+                .applyToServerSettings(builder ->
+                        // see `poolClearedExceptionMustBeRetryable` for the explanation
+                        builder.heartbeatFrequency(50, TimeUnit.MILLISECONDS))
+                .build());
+             FailPoint ignored = FailPoint.enable(failPointDocument, Fixture.getPrimary(client))) {
```

Comment:
I tried passing in `primaryServerAddress` instead of calling `Fixture.getPrimary()` again and the test still passed. Do you have a reason to think it's necessary to call it again with this MongoClient?  The primary should be the same regardless.

Reply:
> I tried passing in primaryServerAddress instead of calling Fixture.getPrimary() again and the test still passed.

It's just to reduce the probability of encountering a race condition described on the new `Fixture.getPrimary` method. However, since I am neglecting the same when calling `FailPoint.enable` from a listener (just to avoid dealing with `InterruptedException` from `Fixture.getPrimary`  in `CommandListener.commandSucceeded`), there is no sense to start caring about it in a different place in the test. I'll change this.

Reply:
Done.